### PR TITLE
long term goals

### PR DESCRIPTION
**Pull request by dakshbhardwaj**
<image src="https://user-images.githubusercontent.com/22423684/180620371-df1ef151-986e-4aee-9ddf-4da99158e478.png" width=200 height="400" />

### CODE REVIEW

File: src/components/GoalCard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { Text, View } from 'react-native';
+import Strings from '../../i18n/en';
+import ProgressBar from '../ProgressBar';
+import { GoalCardStyles } from './GoalCardStyle';
+
+export type GoalCardProps = {
```
Comment (ivinayakg) [inline src/components/GoalCard/index.tsx:7]: Can we create a file which can host all the types and from there import it into this file.

File: src/components/GoalCard/index.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  title: String;
+  taskAssignee: String;
+  taskDescription: String;
+  progress: number;
+};
+
+function GoalCard(props: GoalCardProps) {
+  const styles = GoalCardStyles;
```
Comment (ivinayakg) [inline src/components/GoalCard/index.tsx]: We are importing the GoalCardStyles , so just use that if not then can also change it with importing GoalCardStyles as styles.

File: src/components/GoalScreenHeader/index.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { Image, Text, TouchableOpacity, View } from 'react-native';
+import Images from '../../constants/images/Image';
+import { GoalScreenHeaderStyle } from './GoalScreenHeaderStyle';
+
+export type GoalScreenHeaderProps = {
```
Comment (ivinayakg) [inline src/components/GoalScreenHeader/index.tsx:6]: Again types in a separate file, if can Or even this is good enough as it's exclusive to this component. Your choice

File: src/components/LongGoalsComponent/Data.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
+const data = [
+  {
+    id: '1',
```
Comment (ivinayakg) [inline src/components/LongGoalsComponent/Data.ts]: I'm not aware of what kind of id string is being used at the database wether it is "id" Or "_id" So just make sure to keep it in sync if in future you want to get this data from the backend.

File: src/components/ProgressBar/index.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+const ProgressBarInternal = (props: ProgressBarProps) => {
+  const translateX = useRef(new Animated.Value(0));
+  const styles = ProgressBarStyles;
+
+  useEffect(() => {
+    Animated.timing(translateX.current, {
+      toValue: props.progress / 100,
+      easing: Easing.inOut(Easing.ease),
+      useNativeDriver: false,
```
Comment (ivinayakg) [inline src/components/ProgressBar/index.tsx:16]: I have zero experience with react-native so i dont know how this works, just make sure to cleanup any effects if this needs to.

File: src/components/LongGoalsComponent/Data.ts (hunk 4 above)
Comment (dakshbhardwaj) [reply src/components/LongGoalsComponent/Data.ts]: > I'm not aware of what kind of id string is being used at the database wether it is "id" Or "_id" So just make sure to keep it in sync if in future you want to get this data from the backend.

 I have just followed the format that is already used in the project. I'll do the changes

File: src/components/GoalCard/index.tsx (hunk 1 above)
Comment (dakshbhardwaj) [reply src/components/GoalCard/index.tsx:7]: I think we can keep it here as it is exclusive to the component.

File: src/components/ProgressBar/index.tsx (hunk 5 above)
Comment (dakshbhardwaj) [reply src/components/ProgressBar/index.tsx:16]: nice point, added cleanup function

File: src/components/GoalScreenHeader/index.tsx (hunk 3 above)
Comment (dakshbhardwaj) [reply src/components/GoalScreenHeader/index.tsx:6]: keeping it here only as it's exclusive to the component

File: src/components/GoalScreenHeader/index.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  title: String;
+  onPress: Function;
+  shouldShowArrowDownIcon: Boolean;
+};
+
+function GoalScreenHeader(props: GoalScreenHeaderProps) {
```
Comment (vinayak-trivedi) [inline src/components/GoalScreenHeader/index.tsx:12]: The name could be changed from goalScreenheader to longTermSectionHeader

Comment (dakshbhardwaj) [reply src/components/GoalScreenHeader/index.tsx:12]: I have keep it as goalScreenHeader as  name to make it a reusable component common to any other component that may be introduced in the future on GoalScreen. as it accepts name ,  callback on other function as props.

Comment (vinayak-trivedi) [reply src/components/GoalScreenHeader/index.tsx:12]: ok got it.

**Review by shreya-mishra — COMMENTED** [review]
lgtm :)

**Review by shreya-mishra — APPROVED** [review]

**Pull request merged**